src/masternodes/loan.cpp

```diff
@@ -227,11 +240,29 @@ CAmount CeilInterest(const base_uint<128>& value, uint32_t height)
     return value.GetLow64();
 }
 
-base_uint<128> TotalInterestCalculation(const CInterestRateV2& rate, uint32_t height)
-{
-    auto interest = rate.interestToHeight + ((height - rate.height) * rate.interestPerBlock);
+CNegativeInterest TotalInterestCalculation(const CInterestRateV3& rate, uint32_t height)
+{
+    CNegativeInterest interest;
+    if (rate.interestPerBlock.negative) {
+        // TODO this formula is inaccurate, it assumes that the base amount will remain the same,
+        // when with a negative interest rate the loan amount should reduce on each block. Required
+        // is a formula to reduce the loan amount on each height by the negative interest per block,
+        // and then update the interest per block for the next height, repeat until we are at the
+        // current height. This requires that the loan amount be passed to this function which was
+        // until now not required as the loan amount was static before and so therefore was the interest.
```

Comment:
noob question: why? positive interest didn't increase the base amount, so why should negative interest decrease it?
why not keep it simple and have negative interest in the vault. if `loan + interest <= 0` the loan is removed, so only this check is necessary, but can be done outside.

Reply:
Firstly interests are reduced and only after interests get to 0 loan starts reducing. This cannot happen with positive interest as loan and interests of the loan are considered differently. 

Examples:

- Positive interest

>  Lets asume +0,1 positive interest, then;
>  Loan amount -> 1Token
>  After 1 block -> Loan amount = 1Token; Interest of loan = 0,1Token; Total loan = 1,1Token

- Negative interest

> Asume -0,1 positive interest in a vault with already 0,2Token of interest accrued in the past
> Loan amount -> 1Token; Total loan -> 1,2Token
> After 1 block -> Loan amount = 1Token; Interest of loan = 0,2-0,1Token = 0,1Token; Total loan = 1,1Token
> After 1 block -> Loan amount = 1Token; Interest of loan = 0,1-0,1Token = 0Token; Total loan = 1Token
> After 1 block -> Loan amount = 1-0,1Token = 0,9Token; Interest of loan = 0Token; Total loan = 0,9Token
> After 1 block -> Loan amount = 0,9-0,1Token = 0,8Token; Interest of loan = 0Token; Total loan = 0,8Token

Reply:
Will write a set of tests illustrating the above

Reply:
imho the interest value should become negative and not the loan value be reduced. in the positive interest rate we also don't increase the loan when interest accumulates.
Would also be far easier for users to understand when loan value stays the same (unless there is a payback or takeloan) but interest value increases or decreases (or even goes negative).

(hmm i got notification of a comment, but github is not showing it. my internet bad?)